scripts/gui/CourseDisplay.lua

```diff
@@ -259,22 +259,36 @@ function SimpleCourseDisplay:updateChangesBetween(firstIx, secondIx)
 end
 
 --- Changes the visibility of the course.
-function SimpleCourseDisplay:updateVisibility(visible, onlyStartStopVisible)
+function SimpleCourseDisplay:updateVisibility(visibilityData)

```

Comment:
I would prefer keeping the original boolean parameters (maybe rename visible to allVisible) and adding a onlyCurrentVisible (not last, as that may also mean the last waypoint of the course) and a currentWaypointIx or so parameter.

Reply:
The reason why ive done this is to make it easier to maintain and also add new visibility modes.
This is possible due to having all logic in one place instead of at every function call. 
Also with the packed data inside the table it is easy to add new parameters without having to adjust the whole declaration  of the function.

Reply:
The idea here is that the SimpleCourseDisplay class only cares about displaying the waypoints and nothing else (loose coupling), and so, does not have any dependency on some settings or, for that matter if the vehicle is controlled or not. It is dumb, we have to tell it only which waypoints (if any) to show. 

So yes, I'd still prefer keeping the decision which waypoints to show in the caller, and pass on very simple instructions to this class only.

Reply:
So if I write one centralized function in the course manager that contains all the logic and then passes the logical values through to the simpleCourseDisplay this would be fine, am I right? 

So we have a centralized logic to make maintainance and maybe new visibilityModes easier to add, but still keep the loose coupling, which is in my eyes the best of both worlds.

Reply:
That sounds ok to me, thanks.